### PR TITLE
rafthttp: snapshot sender cleanup

### PR DESCRIPTION
**Pull request by xiang90**
/cc @gyuho

### CODE REVIEW

**Comment by gyuho** [comment]
LGTM!

File: pkg/ioutil/readcloser.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ioutil
+
+import "io"
+
+// ReadCloser implements io.ReadCloser interface.
+type ReadCloser struct {
```
Comment (gyuho) [inline pkg/ioutil/readcloser.go]: Maybe `ReaderCloser` to differentiate `ReadCloser` interface in std package?

Comment (xiang90) [inline pkg/ioutil/readcloser.go]: Probably ReaderAndCloser?

Comment (gyuho) [inline pkg/ioutil/readcloser.go]: Yeah that's even better. Thanks

**Comment by xiang90** [comment]
@gyuho PTAL

**Comment by gyuho** [comment]
LGTM.

**Pull request merged**